pkg/metadata/scheduler.go

```diff
@@ -51,6 +52,9 @@ func (c *Scheduler) Stop() {
 	for _, h := range c.healthHandles {
 		h.Deregister()
 	}
+	for _, stopChannel := range c.stopChannels {
+		close(stopChannel)
```

Comment:
Well, just a nitpick here but given that reading from a closed channel is non-blocking it is likely but not *certain* that these go routines have exited when `Stop()` returns. They may exit on the next tick or later. Using buffered channels would be the way to make sure of that.

Reply:
Having more context on the two options, I'll use that one (the initial) which will guarantee that the go routines has exited after the call.

Reply:
Sorry I mean unbuffered (blocking) channels.